### PR TITLE
Rename confusing variable names in `pl-number-input` controller and mustache files

### PR DESCRIPTION
**Pull request by Frozemint**
As mentioned in https://github.com/PrairieLearn/PrairieLearn/pull/8435#discussion_r1388430939, some of the variable names in the `pl-number-input` controller and mustache files are pretty confusing. To quote @\nwalters512:
> `a_tru` and `a_sub` are almost incomprehensible if you're not already familiar with them. AFAICT there's no reason to not just use `correct_answer` and `submitted_answer`. 

This PR renames the variables `a_tru` and `a_sub` (in the python controller file and mustache file) into `correct_answer` and `submitted_answer` respectively.

### CODE REVIEW

File: apps/prairielearn/elements/pl-number-input/pl-number-input.py [hunk 1]
```diff
@@ -283,13 +283,13 @@ def render(element_html: str, data: pl.QuestionData) -> str:
...
-            a_sub = data["submitted_answers"].get(name)
+            submitted_answer = data["submitted_answers"].get(name)
             # If answer is in a format generated by pl.to_json, convert it
             # back to a standard type (otherwise, do nothing)
-            a_sub = pl.from_json(a_sub)
+            submitted_answer = pl.from_json(submitted_answer)
 
             html_params["suffix"] = suffix
-            html_params["a_sub"] = ("{:" + custom_format + "}").format(a_sub)
+            html_params["a_sub"] = ("{:" + custom_format + "}").format(submitted_answer)
```
Comment (nwalters512) [inline apps/prairielearn/elements/pl-number-input/pl-number-input.py:292]: Change `a_sub` to `submitted_answer`?

Comment (Frozemint) [reply apps/prairielearn/elements/pl-number-input/pl-number-input.py]: Just fixed [0b8c23c](https://github.com/PrairieLearn/PrairieLearn/pull/8846/commits/0b8c23c2246249ff9acf522651675ea566c82583), my bad; missed that one.

File: apps/prairielearn/elements/pl-number-input/pl-number-input.mustache
Comment (nwalters512) [inline apps/prairielearn/elements/pl-number-input/pl-number-input.mustache:1]: I see a reference to `a_sub` in this file that I believe should now be `submitted_answer`.

**Review by nwalters512 — COMMENTED** [review]
How much manual testing has been done so far?

**Comment by Frozemint** [comment]
I have tested this briefly using the template questions and running tests in PL. Is there anything you want to test in particular?

**Review by nwalters512 — APPROVED** [review]
> I have tested this briefly using the template questions and running tests in PL. Is there anything you want to test in particular?

Nothing specific; it's just FUD that there are more bugs lurking after my reviews. Keep in mind that elements don't have particularly robust tests for what they actually render; the tests mostly check parsing/grading behavior. So, manual testing is generally required. I'll approve under the assumption that you've tested enough to be reasonably confident that everything is working as expected now. Let me know when you're ready for me to merge this!

**Comment by Frozemint** [comment]
Since a big change is to the mustache file (renaming submitted_answer and true_answer), I've double checked with the template and element question to make sure submitted and true answers, and it looks like everything shows correctly. 

I've also ran the python tests as newly written in #8435, since we also renamed variables in the precision check functions, and if anything is broken the tests should catch something. But it looks like nothing is broken there as well.

Finally, I've looked at the file diff and it looks like nothing is amiss. I'm fairly confident that there shouldn't be any bugs hiding, but if anyone also wants to chime in (something no bugs can hide with sufficient eyeballs), please do feel free, it'll be greatly appreciated.

**Comment by nwalters512** [comment]
Thank you! I'm confident that between the two of us looking at this carefully, there shouldn't be any more surprises. I'll queue this up to be merged.

**Pull request merged**